### PR TITLE
use LazyUser in SetupManager

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: lib/private/Files/SetupManager.php [hunk 1]
```diff
@@ -530,11 +531,9 @@ private function listenForNewMountProviders() {
 				IMountProvider $provider
 			) {
 				foreach ($this->setupUsers as $userId) {
-					$user = $this->userManager->get($userId);
-					if ($user) {
-						$mounts = $provider->getMountsForUser($user, Filesystem::getLoader());
-						array_walk($mounts, [$this->mountManager, 'addMount']);
-					}
+					$user = new LazyUser($userId, $this->userManager);
+					$mounts = $provider->getMountsForUser($user, Filesystem::getLoader());
```
Comment (PVince81) [inline lib/private/Files/SetupManager.php:535]: what happens if the user does not exist ? does it fail in a clean way wherever it's used inside that function ?

**Review by PVince81 — APPROVED** [review]
:+1: otherwise

File: lib/private/Files/SetupManager.php (hunk 1 above)
Comment (icewind1991) [reply lib/private/Files/SetupManager.php:535]: The user here is one that already had his filesystem setup previously, so any existence check is done previously

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Failures seem related:

OC\User\LazyUser::getUser(): Return value must be of type OCP\IUser, null returned

/home/runner/work/server/server/lib/private/User/LazyUser.php:46
/home/runner/work/server/server/lib/private/User/LazyUser.php:78
/home/runner/work/server/server/lib/private/Files/Storage/Home.php:52
/home/runner/work/server/server/lib/private/Files/Mount/MountPoint.php:167
/home/runner/work/server/server/lib/private/Files/Mount/MountPoint.php:192
/home/runner/work/server/server/lib/private/Files/Filesystem.php:321
/home/runner/work/server/server/lib/private/Files/View.php:227
/home/runner/work/server/server/lib/private/Files/View.php:1859
/home/runner/work/server/server/apps/dav/lib/Connector/Sabre/Directory.php:437
/home/runner/work/server/server/apps/dav/tests/unit/Connector/Sabre/DirectoryTest.php:425
/home/runner/work/server/server/apps/dav/tests/unit/Connector/Sabre/DirectoryTest.php:374

**Comment by blizzz** [comment]
☝

**Comment by szaimen** [comment]
@icewind1991 please fix the conflicts and tests

**Pull request closed without merging**